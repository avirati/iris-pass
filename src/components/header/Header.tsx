import React from 'react'

import { Flex, Link } from 'shared-components'
import { AddEntryButton } from 'components/add-entry-button'

export const Header: React.FC = () => {
  return (
    <Flex
      css={{
        height: '$6',
        backgroundColor: '$tonal600',
        alignItems: 'center',
        px: '$5',
        justifyContent: 'space-between',
        position: 'sticky',
        top: 0
      }}
    >
      <Link
        href="/#/"
        size="lg"
        css={{
          color: '$tonal100',
          textTransform: 'uppercase',
          '&:hover, &:active, &:focus': {
            color: '$tonal100',
            textDecoration: 'none',
          }
        }}
      >
        Password Manager
      </Link>
      <AddEntryButton />
    </Flex>
  )
}