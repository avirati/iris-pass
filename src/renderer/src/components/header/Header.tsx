import React from 'react';
import { ArrowLeft, Rotate, LockAlt, Qr } from '@atom-learning/icons';
import { useHistory, useLocation } from 'react-router-dom';
import { App as CapacitorApp } from '@capacitor/app';

import {
  Flex,
  Heading,
  Icon,
  keyframes,
  styled,
} from '../../shared-components';
import { DarkActionIcon } from '../form-fields';
import { usePasswordSync } from '../../hooks/use-password-sync';
import { useMasterPassword } from '../../hooks/use-master-password';
import { useDevice } from '../../hooks/use-device';

const rotatingAnimation = keyframes({
  from: { transform: 'rotate(0deg)' },
  to: { transform: 'rotate(360deg)' },
});

const StyledActionIcon = styled(DarkActionIcon, {
  '& > svg': { size: '32px' },
  variants: {
    loading: {
      true: {
        animation: `${rotatingAnimation} 2s linear infinite`,
      },
    },
  },
});

export const Header: React.FC = () => {
  const { isAndroid, isElectron } = useDevice();
  const { isUserAuthenticated, lock } = useMasterPassword();
  const { isSyncing } = usePasswordSync();
  const location = useLocation();
  const history = useHistory();
  const isHome = location.pathname === '/';

  React.useEffect(() => {
    if (isAndroid) {
      CapacitorApp.removeAllListeners();
      CapacitorApp.addListener('backButton', () => {
        if (!isHome) {
          history.goBack();
        } else {
          CapacitorApp.exitApp();
        }
      });
      CapacitorApp.addListener('pause', lock);
    }
  }, [history, isAndroid, isHome, lock]);

  return (
    <Flex
      css={{
        height: '$6',
        backgroundColor: '$tonal600',
        alignItems: 'center',
        px: '$5',
        justifyContent: 'space-between',
        position: 'sticky',
        top: 0,
      }}
    >
      <Flex
        css={{
          display: 'flex',
          alignItems: 'center',
          '&:hover, &:active, &:focus': {
            textDecoration: 'none',
          },
        }}
      >
        {isHome ? (
          <Icon is={LockAlt} css={{ color: '$tonal100', mr: '$2' }} />
        ) : (
          <StyledActionIcon
            size='md'
            label='go-back'
            css={{ color: '$tonal100', mr: '$2', border: 'none', size: '24px' }}
            onClick={() => history.goBack()}
          >
            <Icon is={ArrowLeft} />
          </StyledActionIcon>
        )}
        <Heading
          as='h1'
          size='xs'
          css={{
            color: '$tonal100',
            textTransform: 'uppercase',
          }}
        >
          Password Manager
        </Heading>
      </Flex>
      {isUserAuthenticated && (
        <StyledActionIcon
          label='sync'
          loading={isAndroid && isSyncing}
          onClick={() =>
            history.push(isElectron ? '/qrcode/viewer' : '/qrcode/scanner')
          }
          css={{ border: 'none', size: '24px' }}
        >
          <Icon is={isElectron ? Qr : Rotate} />
        </StyledActionIcon>
      )}
    </Flex>
  );
};
