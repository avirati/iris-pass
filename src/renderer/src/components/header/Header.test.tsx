import { render } from '@testing-library/react'

import { Header } from './Header'

describe('Header Component', () => {
  it('renders', () => {
    const { container } = render(<Header />)
    expect(container).toMatchSnapshot()
  })
})