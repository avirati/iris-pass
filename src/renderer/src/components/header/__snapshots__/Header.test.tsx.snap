// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders 1`] = `
<div>
  <div
    class="c-dhzjXW c-dhzjXW-ikodkTj-css"
  >
    <a
      class="c-kgcOue c-kgcOue-jvTRJO-size-lg c-kgcOue-ijmdcqB-css"
      href="/#/"
    >
      Password Manager
    </a>
    <a
      class="c-fkUJsw c-fkUJsw-eYnvrx-size-sm c-fkUJsw-bznQva-cv c-fkUJsw-ijkLaoV-css"
      href="/#/password-form"
    >
      <svg
        aria-hidden="true"
        class="c-dbrbZt c-dbrbZt-hemXWm-size-md c-dbrbZt-ifinTyY-css"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20 12H4m8-8v16"
        />
      </svg>
    </a>
  </div>
</div>
`;
